cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -453,15 +453,21 @@ func isReservationNotReady(errorCode, errorMessage string) bool {
 }
 
 func isInvalidReservationError(errorCode, errorMessage string) bool {
-	reservationErrors := []string{
-		"Incompatible AggregateReservation VMFamily",
-		"Could not find the given reservation with the following name",
-		"must use ReservationAffinity of",
-		"The reservation must exist in the same project as the instance",
-		"only compatible with Aggregate Reservations",
+	regexReservationErrors := []*regexp.Regexp{
```

Comment:
This list is static. Can we compile the regexps once, rather than on every call to this function?

Reply:
Moved the regexp slice to a `var` block.